bot.py

```diff
@@ -448,27 +479,53 @@ async def run_pairing():
 
 async def pair(guild_id: int, timeblock: Timeblock):
     try:
+        # General pairings
         userids = db.query_timeblock(guild_id, timeblock)
         users = [client.get_user(userid) for userid in userids]
         # Users may leave the server without unsubscribing
         # TODO: listen to that event and drop them from the table
         users = list(filter(None, users))
         await process_pairings(guild_id, users, timeblock, is_leetcode=False)
 
-        for difficulty in ["Easy", "Medium", "Hard"]:
-            leetcode_userids = leetcode_db.query_timeblock_difficulty(
-                guild_id, timeblock, difficulty
-            )
-            leetcode_users = [client.get_user(userid) for userid in leetcode_userids]
-            leetcode_users = list(filter(None, leetcode_users))
-            await process_pairings(
-                guild_id,
-                leetcode_users,
-                timeblock,
-                is_leetcode=True,
-                difficulty=difficulty,
+        # Fetch all leetcode subscribers for the timeblock
+        all_leetcode_users = {}
+        with closing(leetcode_db.con.cursor()) as cur:
```

Comment:
this should be a function in db.py. we shouldn't directly interact with sqlite in this file.